span.go

```diff
@@ -52,6 +53,24 @@ type Span interface {
 	// See LogData for semantic details.
 	Log(data LogData)
 
+	// Return a PropagationInjector for the given `format` value, or nil if the
+	// Span does not support such a format.
+	//
+	// OpenTracing defines a common set of `format` values (see
+	// BuiltinPropagationFormat), and each has an expected carrier type.
+	//
+	// Other packages may declare their own `format` values, much like the keys
+	// used by the `net.Context` package (see
+	// https://godoc.org/golang.org/x/net/context#WithValue).
+	//
+	// Example usage (sans error handling):
+	//
+	//     span.PropagationInjectorForFormat(
+	//         opentracing.PROPAGATION_FORMAT_GO_HTTP_HEADER).InjectSpan(
+	//         span, httpReq.Header)
+	//
+	PropagationInjectorForFormat(format interface{}) PropagationInjector
```

Comment:
I know that naming wasn't the main objective of this PR, but to make it less scary I think a simple `Injector(format)` signature achieves the same results..